rts/motoko-rts/src/types.rs

```diff
@@ -572,6 +626,22 @@ impl Array {
     pub unsafe fn len(self: *const Self) -> u32 {
         (*self).len
     }
+
+    pub unsafe fn base_tag(self: *const Self) -> Tag {
+        base_array_tag((*self).header.tag)
+    }
+
+    pub unsafe fn get_slice_start(self: *const Self) -> (Tag, u32) {
+        slice_start((*self).header.tag)
+    }
+
+    pub unsafe fn set_slice_start(self: *mut Self, array_tag: Tag, start: u32) -> () {
+        (*self).header.tag = slice_tag(array_tag, start)
+    }
+
+    pub unsafe fn restore_tag(self: *mut Self, array_tag: Tag) -> () {
+        (*self).header.tag = array_tag;
```

Comment:
I assume the restored tag would fulfil `array_tag == TAG_ARRAY_I || arraty_tag == TAG_ARRAY_M || array_tag == TAG_ARRAY_T` (no slicing any longer)? Not sure, if a debug assertion would make sense.

Reply:
added assert